nexus/src/external_api/params.rs

```diff
@@ -177,6 +177,8 @@ pub struct DiskCreate {
     pub snapshot_id: Option<Uuid>, /* TODO should be a name? */
     /** size of the Disk */
     pub size: ByteCount,
+    /** if present, on-disk blocks should be encrypted */
+    pub encryption_key: Option<Uuid>,
```

Comment:
So the type is a UUID, but what values should we actually be passing here?

From looking at the `openapi/nexus.json` file, it's clear that this is plumbed through the external API, but if I was a client of the Oxide API, I'd have no idea what values are valid to be used.

Is there another API where you expect callers to add encryption keys before they can start using encrypted disks? Is there some set of UUIDs that we'll make valid on the Oxide side of things?

Reply:
> Is there another API where you expect callers to add encryption keys before they can start using encrypted disks? Is there some set of UUIDs that we'll make valid on the Oxide side of things?

I originally had this as an "encrypted" boolean, where the client would simply have to decide "encrypted or not", but changed it for this PR to expect what you're describing: a future endpoint where clients create keys, and choose which keys are used for which disks.

Now that I'm typing it all out, it may be better *not* to offer the choice of clients managing keys and which keys map to which internal consumers (i.e. disks), and to simply ask if users want on-disk blocks to be encrypted or not. That way the implementation details are left up to Nexus.

I'm also not sure if there are previous determinations on this topic or not. I couldn't find any existing encryption key code.

Reply:
I think we should dig into this, especially if we want to ensure "encrypted by default" remains easy.

If you want to punt the decision, and just bump the revision of Crucible in this PR, I think I'd be okay passing a `None` encryption key to Crucible internally. However, I'd want to avoid changing the external API until we need to - otherwise, console + friends will need to start plumbing this UUID field that we might not want.

Does that seem reasonable?

Reply:
> Just for additional context, support for customer-controlled keys was an explicit non-requirement from [RFD 29](https://rfd.shared.oxide.computer/rfd/0029).

Thanks, I missed this determination :( I'll update the PR.